src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using FluentAssertions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Steeltoe.Common.Options;
+using Steeltoe.Common.Utils.IO;
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Steeltoe.Common.Security.Test;
+
+public class CertificateRotationServiceTest
+{
+    [Fact]
+    [Trait("Category", "SkipOnMacOS")]
+    public async Task ServiceLoadsCertificate()
+    {
+        var personalCertStore = new X509Store(StoreName.My, StoreLocation.CurrentUser);
+        personalCertStore.Open(OpenFlags.ReadWrite);
+        personalCertStore.Certificates.Find(X509FindType.FindByIssuerName, "Diego Instance Identity Intermediate CA", false).Should().BeEmpty();
+
+        using var sandbox = new Sandbox();
+        var filename = sandbox.CreateFile("fakeCertificate.p12");
+        File.Copy("instance.p12", filename, true);
+
+        var config = new ConfigurationBuilder()
+            .AddCertificateFile(filename)
+            .Build();
+        var services = new ServiceCollection();
+        services.AddSingleton<IConfiguration>(config);
+        services.AddOptions();
+        services.AddSingleton<IConfigureOptions<CertificateOptions>, ConfigureCertificateOptions>();
+        services.AddSingleton<ICertificateRotationService, CertificateRotationService>();
+
+        var serviceProvider = services.BuildServiceProvider();
+        var service = serviceProvider.GetRequiredService<ICertificateRotationService>();
+
+        service.Start();
+
+        var collection = personalCertStore.Certificates.Find(X509FindType.FindByIssuerName, "Diego Instance Identity Intermediate CA", false);
+        collection.Should().NotBeNull();
+
+#if NET6_0_OR_GREATER
+        if (!File.Exists(Path.Combine(LocalCertificateWriter.AppBasePath, "GeneratedCertificates", "SteeltoeInstanceCert.pem")))
```

Comment:
Why check if the file exists? It uses the sandbox, so I would expect it to either always or never exist.

Reply:
This portion of the test is not using the sandbox. These certificates _may_ be generated by another test that has already run, or they may not be present yet. Ultimately, I was looking for an easy path to get valid certificate data that did not exactly match the .p12 certificate (I think the .pem files that are committed to source right next to it have the same content) so that we can detect changes. I don't feel strongly in either direction about keeping this check.